tests/functional/aws-node-sdk/test/object/initiateMPU.js

```diff
@@ -65,5 +66,118 @@ describe('Initiate MPU', () => {
                 done();
             });
         });
+
+        describe('with tag set', () => {
+            it('should be able to put object with 10 tags',
+            done => {
+                const taggingConfig = generateMultipleTagQuery(10);
+                s3.createMultipartUpload({
+                    Bucket: bucket,
+                    Key: key,
+                    Tagging: taggingConfig,
+                }, err => {
+                    assert.ifError(err);
+                    done();
+                });
+            });
+
+            it('should allow putting 50 tags', done => {
+                const taggingConfig = generateMultipleTagQuery(50);
+                s3.createMultipartUpload({
+                    Bucket: bucket,
+                    Key: key,
+                    Tagging: taggingConfig,
+                }, err => {
+                    assert.ifError(err);
+                    done();
+                });
+            });
+
+            it('should return BadRequest if putting more that 50 tags',
+            done => {
+                const taggingConfig = generateMultipleTagQuery(51);
+                s3.createMultipartUpload({
```

Comment:
Is this a promise? I would use assert.rejects if so. Assert.throws would be good if it threw an error instead of just returning it.

Reply:
Tis not a promise

Reply:
dang